src/java.base/share/classes/java/lang/foreign/MemoryLayout.java

```diff
@@ -709,10 +709,7 @@ static SequenceLayout sequenceLayout(MemoryLayout elementLayout) {
      */
     static StructLayout structLayout(MemoryLayout... elements) {
         Objects.requireNonNull(elements);
-        return wrapOverflow(() ->
-                StructLayoutImpl.of(Stream.of(elements)
-                        .map(Objects::requireNonNull)
-                        .toList()));
+        return wrapOverflow(() -> StructLayoutImpl.of(List.of(elements)));
```

Comment:
Why was Objects.requireNonNull dropped? I believe that NPE  will be thrown deeper inside the implementation - which is why we had the eager check (to make stack trace more readable).